### PR TITLE
Remove default values for consumer_group from offset commit/fetch structs

### PR DESCRIPTION
**Pull request by dantswain**
This was masking two bugs - see the changes to server.ex

Also includes a tiny test edge case fix.

### CODE REVIEW

File: test/integration/consumer_group_test.exs [hunk 1]
```diff
@@ -51,7 +51,7 @@ defmodule KafkaEx.ConsumerGroup.Test do
                           uris: Application.get_env(:kafka_ex, :brokers),
                           consumer_group: consumer_group)
 
-    offset_before = TestHelper.latest_offset_number(topic, 0, worker_name)
+    offset_before = TestHelper.latest_offset_number(topic, 0, worker_name) || 0
```
Comment (bjhaid) [inline test/integration/consumer_group_test.exs]: This should live in `TestHelper.latest_offset_number`

Comment (bjhaid) [inline test/integration/consumer_group_test.exs]: Fixing this along with the merge, thanks!
